src/CMake/PluginMacros.cmake

```diff
@@ -215,5 +215,31 @@ FUNCTION(ADD_DATABASE_CODE_GEN_TARGETS gen_name)
 
         add_dependencies(gen_plugin_all ${gen_target_name})
     endif()
-ENDFUNCTION(ADD_DATABASE_CODE_GEN_TARGETS)
+endfunction()
+
+
+macro(VISIT_SELECTED_PLUGIN_ERROR type plist msg)
```

Comment:
I am on the fence about functiona pre-ambles especially for those functions that are easy enough to understand the code itself kinda sorta speaks for itself. But, does this, and the one below, need a pre-amble?

One line that confuses me quite a bit is 225 `if(NOT foundPlug EQUAL -1)` which suggests the value for `foundPlug` that indicates *success* is `-1` which is fine if thats the way it is but it just seems a bit odd. I was expecting `1` or `true` or something.

Reply:
list(FIND ... ) returns -1 if what is being searched for isn't found. Otherwise it returns an index into the list.

Reply:
Yes, the if test seems funky, but its actually testing whether or not the passed in plist (list of plugins that cannot be built, generally due to missing dependencies) are part of the SELECTED list.  We want an error condition and not just a status message in that case, because user specifically requested that plugin be built.